Add VLAN enhancement documentation (#216)
harvester/docs: docs/networking/clusternetwork.md

---
sidebar_position: 1
sidebar_label: Cluster Network
title: ""
keywords:
- Harvester
- Networking
- ClusterNetwork
- NetworkConfig
- Network
---
# Harvester Networking

## Concepts

### Cluster Network

> 
> We bring in a new concept called cluster network for traffic isolation.

Reviewer: ```suggestion
_Available as of v1.1.0_

In Harvester v1.1.0, we introduced a new concept called cluster network for traffic isolation.
```

The following diagram describes a typical network architecture that separates data-centre (DC) traffic from out-of-band(OOB) traffic.
![](/img/v1.1/networking/traffic-isolation.png)

We abstract the sum of devices, links, and configurations on a traffic-isolated forwarding path on the Harvester as a cluster network.

In the above case, there will be two cluster networks corresponding to two traffic-isolated forwarding paths.

### Network Configuration

> The specifications including network devices of the Harvester hosts can be different. To be compatible with such a heterogeneous cluster, we design network configuration.

Reviewer: ```suggestion
Specifications including network devices of the Harvester hosts can be different. To be compatible with such a heterogeneous cluster, we designed the network configuration.
```

> The network configuration can only work under a certain cluster network. Each network configuration corresponds to a set of hosts with uniform network specifications. Therefore, it requires multiple network configurations to enable a cluster network on multiple non-uniform hosts.

Reviewer: ```suggestion
Network configuration only works under a certain cluster network. Each network configuration corresponds to a set of hosts with uniform network specifications. Therefore, multiple network configurations are required for a cluster network on non-uniform hosts.
```

### Network

> The network is an interface used in VMs to connect with the host network. Same with network configuration, every network must be under a certain cluster network except the built-in network.

Reviewer: ```suggestion
A network is an interface in a virtual machine that connects to the host network. As with network configuration, every network except the built-in `mgmt` network must be under some cluster network.
```

## Concepts

### Cluster Network
_Available as of v1.1.0_

In Harvester v1.1.0, we introduced a new concept called cluster network for traffic isolation.

The following diagram describes a typical network architecture that separates data-center (DC) traffic from out-of-band (OOB) traffic.
![](/img/v1.1/networking/traffic-isolation.png)

We abstract the sum of devices, links, and configurations on a traffic-isolated forwarding path on Harvester as a cluster network.

In the above case, there will be two cluster networks corresponding to two traffic-isolated forwarding paths.

### Network Configuration

Specifications including network devices of the Harvester hosts can be different. To be compatible with such a heterogeneous cluster, we designed the network configuration.

Network configuration only works under a certain cluster network. Each network configuration corresponds to a set of hosts with uniform network specifications. Therefore, multiple network configurations are required for a cluster network on non-uniform hosts.

### Network

> A network is an interface in a virtual machine that connects to the host network. As with network configuration, every network except the built-in `mgmt` network must be under some cluster network.

Reviewer 2: ```suggestion
A network is an interface in a virtual machine that connects to the host network. Same as network configuration, every network (except the built-in `mgmt` network) must be under some cluster network.
```

Reviewer 2: Does it mean `same as`?
